pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/ComputeValueAndNullTogetherTransformFunction.java

```diff
@@ -0,0 +1,187 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.operator.transform.function;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+import org.apache.pinot.core.operator.ColumnContext;
+import org.apache.pinot.core.operator.blocks.ValueBlock;
+import org.apache.pinot.spi.data.FieldSpec;
+import org.roaringbitmap.RoaringBitmap;
+
+
+/**
+ * Base class for transform functions that compute value and NULL together when NULL handling is enabled.
+ */
+public abstract class ComputeValueAndNullTogetherTransformFunction extends BaseTransformFunction {
```

Comment:
The class name is a little bit confusing. Basically it means the handling with/without null handling is different

Reply:
The reason with/without null handling the result is different is because the value and null are computed together. Or in other words, with/without null handling the result is always the same if value and null are computed separately.

How about we keep the class name and add to the class comment "Subclasses of this class may have different results (value/null bitmap) with/without null handling."?

Reply:
How about adding this to the class comment: "When null handling is enabled, subclasses of this class need to compute using value and null together in order to get correct results."?

Reply:
Gave it more thoughts and renamed it to `ComputeDifferentlyWhenNullHandlingEnabledTransformFunction`.